src/components/MembershipCheckButton.tsx

```
@@ -0,0 +1,86 @@
+    }
+  };
+
+  return (
+    <div className={`text-center ${className || ''}`}>
+      <ErrorPopup error={error} onClose={clearError} />
+      <MembershipStatusPopup status={status} onClose={() => setStatus(null)} />
+      <button
+        onClick={handleMembershipCheck}
+        disabled={isChecking}
+        className="inline-flex w-full items-center justify-center gap-2 rounded-lg bg-tangerine-500 px-8 py-4 text-lg font-bold text-white shadow-lg transition-all hover:bg-tangerine-600 hover:shadow-xl disabled:cursor-not-allowed disabled:opacity-50"
+      >
```

Biome (2.3.13)

[error] Provide an explicit type prop for the button element.

The default type of a button is submit, which causes the submission of a form when placed inside a `form` element. This is likely not the behaviour that you want inside a React application.
Allowed button types are: submit, button or reset

(lint/a11y/useButtonType)